manifests/server/installdb.pp

```diff
@@ -38,6 +37,7 @@
       user                => $mysqluser,
       log_error           => $log_error,
       defaults_extra_file => $_config_file,
+      require             => File[$options['mysqld']['log-error']],
```

Comment:
2/ This `mysql_datadir` resource is always declared with a relationship, so we might end-up with:

```
require => File[undef]
```

when `$options['mysqld']['log-error']` is not set, which seems to trigger your error.